ivanp-epam/nodejs-aws-be
# task-9

# Note: AWS Account has been suspended to prevent money consumption. 

### Services URLS: 
- **BFF_URL**: http://ivanp-epam-bff-api-dev.eu-west-1.elasticbeanstalk.com
- **HTTPS BFF_URL**: https://deyaerotyi.execute-api.eu-west-1.amazonaws.com/dev
- **CART_URL**: "https://6ux44zy8w5.execute-api.eu-west-1.amazonaws.com/dev/api"
- **PRODUCT_URL**: "https://vmle7gk5m6.execute-api.eu-west-1.amazonaws.com/dev"
- **SITE with BFF enabled** https://dvcrugb1m68oa.cloudfront.net/
- **FE PR** https://github.com/ivanp-epam/nodejs-aws-fe/pull/7

### Usage and test examples: 

#### Raw requests:
```
GET http://ivanp-epam-bff-api-dev.eu-west-1.elasticbeanstalk.com/products
GET http://ivanp-epam-bff-api-dev.eu-west-1.elasticbeanstalk.com/profile/cart

POST 'http://ivanp-epam-bff-api-dev.eu-west-1.elasticbeanstalk.com/products' 
  HEADERS 
        content-type: application/json;charset=UTF-8
  DATA 
       {"price":123,"description":"asd","title":"qwe","count":"123"}
```

#### Using FE:
1. Open https://dvcrugb1m68oa.cloudfront.net/
2. Check products
3. Add products (Profile icon -> Manage products -> Create products)
4. Add products to the cart
5. Check Cart by clicking on cark icon

#### [TASK Description](https://github.com/ivanp-epam/nodejs-aws-be/blob/be484769b2a06207788b4d819aeff1bc00339dad/bff-service/README.md)

## Evaluation criteria (each mark includes previous mark criteria)

### Provide your reviewers with the following information:

- [x] link to the repo
- [x] product-service service API endpoint URL
- [x] example of the create product API call with all needed information: URL, payload, headers, etc.
- [x] CART service API endpoint URL
- [x] bff-service service URL
- [x] example how to call product-service and CART services via bff-service service URL

----

- [x] 3 - A working and correct ~~express~~ **nest** application should be in the bff-service folder. Reviewer can start this application locally with any valid configuration in the .env file and this application should works as described in the task 9.1


#### Additional (optional) tasks

    **How to test:**
     * Get products list
     * Create new product
     * Get products list - result shouldn’t have new product
     * Wait more than 2 minutes
     * Get products list - result should have new product
